FIX - Edit propal date

BB2A-Anthony opened the pull request:
# FIX|Fix #[Edit propal date]

if the proposal is validated the date is not editable
Using the new method to edit the date
Updates the validity date of the proposal

eldy commented on htdocs/comm/propal/class/propal.class.php:
```diff
@@ -2060,8 +2060,11 @@ public function set_date($user, $date, $notrigger = 0)
 
-			$sql = "UPDATE ".MAIN_DB_PREFIX."propal SET datep = '".$this->db->idate($date)."'";
-			$sql .= " WHERE rowid = ".((int) $this->id)." AND fk_statut = ".self::STATUS_DRAFT;
+			if (isset($this->duree_validite) && !empty($this->fin_validite)) {
```
set_date is a low level method:
So if we call it, we must update the field.
conditions must be moved on caller side, they will depends on context of this caller page.

bb2a replied:
@eldy moved conditions on caller side

eldy replied:
Oops problem in PR, 474 files are modified...

bb2a replied:
@eldy sorry manipulation error I create another PR #21839

Commit:
NEW : add margin infos to takepos invoice lines

Commit:
add hook to complete payment

frederic34 committed:
doc

eldy committed:
Responsive

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
css

lvessiller-opendsi committed:
NEW show SellBy/EatBy dates for each batch product in shipment card

eldy committed:
Clean deprecated code. Fix go back to mode1 in supplier invoice

eldy committed:
Fix avoid error message when pressing enter into bank reconciliation

eldy committed:
Merge pull request #21584 from Quatadah/template1

NEW: Adding template 01

eldy committed:
Fix lang

eldy committed:
Add version and date of website template generation in the sql.

eldy committed:
Clean code: No more fetch to clone

eldy committed:
Merge pull request #21721 from frederic34/patch-4

doc

eldy committed:
Merge pull request #21713 from grandoc/new_branch_11_08_2022

update code for PHP8 compliance

eldy committed:
Merge pull request #21714 from frederic34/outh2forswiftmailer

token was not refreshing

eldy committed:
Merge pull request #21719 from altairisfr/takeposmargin

NEW : add margin infos to takepos invoice lines

eldy committed:
Merge pull request #21720 from altairisfr/takepospay

NEW : TakePOS - add hook to complete payment

eldy committed:
Merge pull request #21712 from defrance/patch-223

NEW  if we select another view list mode, we keep it

eldy committed:
Merge pull request #21574 from ymollard/create3rdpartyV14

FIX: Create a 3rd party from the self-subscription membership form

eldy committed:
doc

grandoc committed:
update code for PHP8 compliance

grandoc committed:
update code for PHP8 compliance

grandoc committed:
update code for PHP8 compliance

grandoc committed:
update code for PHP8 compliance

grandoc committed:
update code for PHP8 compliance

grandoc committed:
update code for PHP8 compliance

grandoc committed:
update code for PHP8 compliance

atm-florianm committed:
Ticket `oldcopy`: use dol_clone() instead of fetching; keep oldcopy up to date if the methods are called several times

frederic34 committed:
keep batch info when correcting stock

Commit:
Token now refresh before it expire

Commit:
expire was initiated as true instead of false

eldy committed:
Fix trans

eldy committed:
Merge branch '16.0' of git@github.com:Dolibarr/dolibarr.git into develop

Conflicts:
	ChangeLog

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

dolibit-ut committed:
Update and rename dev/resources/iso-normes/world_tax_rates.txt to dev/resources/iso-normes/tax/world_tax_rates.txt

check websites

move to /tax/

lvessiller-opendsi committed:
Merge branch 'develop' of github.com:Dolibarr/dolibarr into new-expedition-card-eat-sell-by-show

dolibit-ut committed:
Update and rename dev/resources/iso-normes/vat_number_names.txt to dev/resources/iso-normes/tax/vat_number_names.txt

add  [de]  label names 
add  [it]    IVA

move to /tax/

lvessiller-opendsi committed:
NEW show sell-by and eat-by dates only if not empty

eldy committed:
Merge branch '16.0' of git@github.com:Dolibarr/dolibarr.git into develop

Conflicts:
	htdocs/langs/fr_FR/accountancy.lang
	htdocs/langs/fr_FR/admin.lang
	htdocs/langs/fr_FR/cashdesk.lang
	htdocs/langs/fr_FR/main.lang
	htdocs/langs/fr_FR/members.lang
	htdocs/langs/fr_FR/products.lang

SylvainLegrand committed:
Fix : improve date library

stickler-ci committed:
Fixing style errors.

Deefaze committed:
Update newonlinesign.php

missing "commercial" language file for complete translation

eldy committed:
Merge branch '16.0' of git@github.com:Dolibarr/dolibarr.git into develop

defrance committed:
php v8 warning

defrance committed:
php V8 warning

defrance committed:
php V8 warning

stickler-ci committed:
Fixing style errors.

defrance committed:
Update mod_propale_saphir.php

eldy committed:
Fix duplicate string

dolibit-ut committed:
Update llx_00_c_country.sql

Cambodge / Cambodia
Central African Republic (CAR/RCA)
Chile
DR Congo (RDC)
Croatia
Vatican City (Saint-Siège)
Indonesia
Jamaica
Japan (Nippon)
North Macedonia

Commit:
add redirect on action confirm addconsumedline and addproduceline

dolibit-ut committed:
Update llx_20_c_departements.sql

defrance committed:
php V8 warning

eldy committed:
Merge pull request #21742 from dolibit-ut/patch-411

Update llx_00_c_country.sql

eldy committed:
Merge pull request #21743 from dolibit-ut/patch-412

Update llx_20_c_departements.sql

eldy committed:
Merge pull request #21745 from defrance/patch-228

php V8 warning

defrance committed:
php V8 warning

eldy committed:
Clean code. Prepare for inputing prices with taxes

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

vdegrandpre committed:
CLOSE 21748 Maillage sur ErrorProductAlreadyExists en créant produit

stickler-ci committed:
Fixing style errors.

Commit:
NEW: Added "Get lines from BOM" at the REST Service

dolibit-ut committed:
Update llx_20_c_departements.sql

Greece

dolibit-ut committed:
Update 16.0.0-17.0.0.sql

sort alphabetical

eldy committed:
Merge pull request #21751 from dolibit-ut/patch-413

Update llx_20_c_departements.sql

eldy committed:
Merge pull request #21752 from dolibit-ut/patch-414

Update 16.0.0-17.0.0.sql

eldy committed:
Merge pull request #21738 from defrance/patch-224

php v8 warning

eldy committed:
Merge pull request #21746 from defrance/patch-229

php V8 warning

eldy committed:
Update mod_propale_saphir.php

eldy committed:
Merge pull request #21741 from defrance/patch-227

php V8 warning

eldy committed:
Merge pull request #21744 from atm-jpb/NEW/redirectOnConfirmLine

NEW : add redirect on action confirm addconsumedline and addproduceline

eldy committed:
Update mod_expensereport_sand.php

eldy committed:
Merge pull request #21739 from defrance/patch-225

php V8 warning

eldy committed:
Merge pull request #21740 from defrance/patch-226

php V8 warning

eldy committed:
Code comment

eldy committed:
Merge pull request #21733 from SylvainLegrand/Fix_improve_date_library

Fix : improve date library

eldy committed:
Merge pull request #21734 from Deefaze/patch-1

FIX : newonlinesign.php missing commercial file in loadings

eldy committed:
Merge pull request #21731 from dolibit-ut/patch-410

Update and rename dev/resources/iso-normes/vat_number_names.txt to de…

eldy committed:
Merge pull request #21729 from dolibit-ut/patch-409

Update and rename dev/resources/iso-normes/world_tax_rates.txt to dev…

eldy committed:
Merge pull request #21727 from fboitel/smtp-by-oauth-refresh-token

smtp by oauth2: token now refresh before it expires

vdegrandpre committed:
commentaires du code en anglais

stickler-ci committed:
Fixing style errors.

eldy committed:
Merge pull request #21635 from atm-gauthier/NEW/conf_to_always_update_thm_of_task_time_line

NEW : conf TIMESPENT_ALWAYS_UPDATE_THM, when it's on we always check …

Commit:
#21750 NEW: Added "Get lines from BOM" at the REST Service

eldy committed:
Merge pull request #21749 from vdegrandpre/develop

CLOSE 21748 Maillage sur ErrorProductAlreadyExists en créant produit

eldy commented on htdocs/comm/propal/card.php:
```diff
@@ -315,6 +315,9 @@
 
 		if (!$error) {
+			if (isset($object->duree_validite) && !empty($object->fin_validite)) {
+				$object->fin_validite = $datep + ($object->duree_validite * 24 * 3600);
```
So you add code to avoid bad sync between duration of validity and end of validity. Why not.
But imagine, we initialy create a proposal the 1st of january with a validity duration of 10 days. So the end of validity is set during creation at 10th of january. Dolibarr is using this date for alerts.
Imagine the customer ask more delay. So the saler increase the end of validity date to 30th of january. But here, the code seems to overwrite this 30th of january to 10th of january. Am i wrong ?

bb2a replied:
yes it is for a change of date. @eldy for you what would be the best solution?  To update the validity date.  take back the gap between the two dates?  or put options or offer to update the date?

SylvainLegrand committed:
Improve settings for phone number (Madagascar)

eldy committed:
FIX empty with hasRight and add phpunit to avoid this in future

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch 'develop' into new_branch_12_08_2022

eldy committed:
Merge pull request #21724 from grandoc/new_branch_12_08_2022

update code for PHP8 compliance

eldy committed:
Clean code

SylvainLegrand committed:
PR new hook "addMoreActionsButtons" on movement list

eldy committed:
Fix phpv8 warnings

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
# WARNING: head commit changed in the meantime

Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into
develop

eldy committed:
Merge branch '16.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Clean warning

eldy committed:
Fix trans

eldy committed:
Missing migration line

eldy committed:
css

eldy committed:
css

Commit:
#21750 NEW: Added "Get lines and Post lines from BOM" at the REST Service

stickler-ci committed:
Fixing style errors.

Commit:
#21750 NEW: Added "Get lines and Post lines from BOM" at the REST Service

Commit:
Merge remote-tracking branch 'origin/15.0' into 15.0

ptibogxiv committed:
Fix updating language thirdparty to member

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into correctstockbatch

eldy committed:
Clean code for http header + better support for Content-Security-Policy

eldy committed:
Fix recommended value

eldy committed:
NEW Add the referrer-policy to "same-origin" by default.

eldy committed:
NEW can set header "Strict-Transport-Security" in web sites

Quatadah committed:
template 04

grandoc committed:
FIX php8 compatibility

eldy committed:
Can add Permissions-Policy on web sites

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

eldy committed:
Merge pull request #21764 from grandoc/new_branch_16_08_2022

FIX php8 compatibility

eldy committed:
Merge pull request #21762 from ptibogxiv/patch-37

Fix updating language thirdparty to member

eldy committed:
Merge pull request #21754 from SylvainLegrand/PR_improve_phone_number_settings_for_madagascar

Improve settings for phone number (Madagascar)

eldy committed:
Merge pull request #21759 from SylvainLegrand/PR_new_hook_on_movement_list

PR new hook "addMoreActionsButtons" on movement list

eldy committed:
Merge pull request #21750 from Humml87/15.0

NEW: Added "Get lines and Post lines from BOM" at the REST Service

eldy committed:
Merge pull request #21763 from Quatadah/dolicloud-qnasdami-template04

#NEW Template 04

Commit:
Fix #21761 Remove dopple +1 on $ranktouse.

Commit:
Fix #21768 The Contact is now showing in the card, if no company is selected.

eldy committed:
NEW Can delete a website

Commit:
Clean Code: Remove duplicate code form bom_card.php

stickler-ci committed:
Fixing style errors.

Commit:
Fix #20004 correct SQL command to detect "products to be ordered"

Humml87 committed:
Merge pull request #1 from Humml87/develop_Clean_dopple_Code

Develop clean dopple code

Humml87 committed:
Merge pull request #2 from Humml87/develop_#20004

Fix #20004 correct SQL command to detect "products to be ordered"

Humml87 committed:
Merge branch 'develop_HUSCH' into develop_#21761

stickler-ci committed:
Fixing style errors.

Commit:
Affichage du split all (numéros de série uniques)

Commit:
Ajout de la quantité maximum par produit

Commit:
FIX data quantités sur tr

Quatadah committed:
reexporting template 03

fcharlaix-opendsi committed:
Add pid integer column to llx_cronjob

Store the PID of the cronjob to allow cleanup if the process is no longer present in the system

Commit:
Traitement du splitall

Commit:
New #21774 extend REST with PUT and DELETE for BOM-Lines. Otherwise some changes for correct ranking of BOM-Lines.

stickler-ci committed:
Fixing style errors.

Commit:
FIX erreur

eldy committed:
Merge pull request #21797 from Humml87/develop_#21774_REST_BOM-Line_PUT_DELETE

New #21774 extend REST with PUT and DELETE for BOM-Lines.

eldy committed:
Merge pull request #21790 from fcharlaix-opendsi/21780-db

CLOSE #21780 Add pid integer column to llx_cronjob

eldy committed:
Merge pull request #21789 from Quatadah/dolicloud-template03

 FIX Template 01 in #21775

eldy committed:
Merge pull request #21782 from Humml87/develop_#20004

Fix #20004 correct SQL command to detect "products to be ordered"

fcharlaix-opendsi committed:
CLOSE #21780 Add pid field to Cronjob class and store PID on job execution

Add the new pid column to the Cronjob class internal value and functions

 Store the current PID on job execution during run_jobs

stickler-ci committed:
Fixing style errors.

Commit:
FIX mode de calcul splitall

Humml87 committed:
Merge branch 'Dolibarr:develop' into develop_#21761

Commit:
Clean

frederic34 committed:
iban human readable

Commit:
Fix #21761 Correct numbering by adding/updating the BomLine and clean code.

frederic34 committed:
iban human readable

frederic34 committed:
Update cashcontrol_card.php

frederic34 committed:
Update codeinit.php

frederic34 committed:
Update codeinit.php

frederic34 committed:
Update barcode.php

frederic34 committed:
use isModEnabled

frederic34 committed:
fix syntax

frederic34 committed:
apply fix

frederic34 committed:
Merge remote-tracking branch 'upstream/develop' into correctstockbatch

Commit:
Fix: correct parameter "backtopage".

ptibogxiv committed:
New can view and set no_email for contacts

stickler-ci committed:
Fixing style errors.

ptibogxiv committed:
Update api_contacts.class.php

Commit:
Fix #21816: correct Agenda filter (if no filter is selected)

Commit:
Fix: left menu can't create. Now the check is correct.

ptibogxiv committed:
Fix hook in takepos.pay.php

ptibogxiv committed:
ADD belgium public holiday

ptibogxiv committed:
Update 16.0.0-17.0.0.sql

dolibit-ut committed:
Update ChangeLog

double entries deleted in ChangeLog 15.0.3

Commit:
Fix: left and top menu can't create. Now the checks are correct. GETPOST with method 3 (POST then GET).

eldy committed:
Fix regression

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge branch '16.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Fix missing llxHeader()

eldy committed:
FIX MVC and missing http header in TakePOS

BB2A-Anthony committed:
Merge branch 'develop' of https://github.com/Dolibarr/dolibarr into FIX---Edit-propal-date-V16

eldy committed:
Merge pull request #21826 from dolibit-ut/patch-415

Update ChangeLog

eldy committed:
Merge pull request #21825 from ptibogxiv/patch-41

ADD belgium public holiday

eldy committed:
Merge pull request #21824 from ptibogxiv/patch-40

Fix hook in takepos/pay.php

eldy committed:
Merge pull request #21817 from Humml87/develop_#21816_Agenda_filter

Fix #21816: correct Agenda filter (if no filter is selected)

eldy committed:
Merge pull request #21815 from ptibogxiv/patch-39

New can view and set no_email for contacts with API

eldy committed:
Merge pull request #21819 from Humml87/develop_Fix_Create_left_menu

Fix: left and top menu can't create. Now the checks are correct.

eldy committed:
Merge pull request #21805 from frederic34/patch-4

Update cashcontrol_card.php

eldy committed:
Merge pull request #21806 from frederic34/patch-5

fix php8 warnings

eldy committed:
Merge pull request #21803 from frederic34/ibanForHuman

iban human readable

Commit:
Fix: add urlencode to backtopage.

eldy committed:
Close #21786 Merged manually to solve conflict on binary file

eldy committed:
Merge branch '16.0' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #21808 from Humml87/develop_Fix_MO_Agenda_Button_backtopage

Fix: Parameter "backtopage" on AddEvent Button

eldy committed:
Merge pull request #21804 from Humml87/develop_#21761

Fix #21761 Correct numbering by adding/updating the BomLine and clean code.

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #21798 from fcharlaix-opendsi/21780

CLOSE #21780 Add pid field to Cronjob class and store PID on job execution

eldy committed:
Merge pull request #21800 from atm-lena/develop_NEW_SplitAll_OF_V2

MO : Split All of v2

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

eldy committed:
Merge pull request #21770 from Humml87/develop_#21768

Fix #21768 The Contact is now showing in the card, if no company is selected

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

grandoc committed:
FIX php8 compatibility

defrance committed:
php V8 warning

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

defrance committed:
Update functions2.lib.php

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

grandoc committed:
FIX php8 compatibility

eldy committed:
Merge pull request #21718 from ATM-Consulting/NEW_add_oldcopy_for_ticket_triggers

NEW: add oldcopy to Ticket for modules intercepting TICKET_MODIFY

eldy committed:
Clean code using of oldcopy

eldy committed:
Merge pull request #21832 from grandoc/new_branch_22_08_2022

FIX php8 compatibility

eldy committed:
Update functions2.lib.php

eldy committed:
Merge pull request #21833 from defrance/patch-235

php V8 warning (supplier order)

eldy committed:
Merge pull request #21726 from frederic34/correctstockbatch

keep batch info when correcting stock

eldy committed:
Merge pull request #21722 from Easya-Solutions/new-expedition-card-eat-sell-by-show

NEW show SellBy/EatBy dates for each batch product in shipment card

eldy committed:
Merge pull request #21673 from bb2a/setup-yesno

Fix - Form Setup yesno

eldy committed:
Merge branch 'develop' of git@github.com:Dolibarr/dolibarr.git into develop

BB2A-Anthony committed:
O
Merge branch '16.0' of https://github.com/Dolibarr/dolibarr into FIX---Edit-propal-date-V16

BB2A-Anthony committed:
ok

BB2A-Anthony committed:
Merge branch 'Dolibarr:develop' into FIX---Edit-propal-date-V16

BB2A-Anthony commented:
PR https://github.com/Dolibarr/dolibarr/pull/21839
